### PR TITLE
9444 push images

### LINKED ISSUE #9444: Push dev app images to registries
- develop branch goes to DockerHub
- each PR branch goes to GitHub image registry

### LINKED ISSUE #9539: Change tag from "stable" to "alpha" on Docker Hub
I'm having second thoughts about using the term "stable" for Docker images: https://preview.guides.gdcc.io/en/develop/container/base-image.html#supported-image-tags

Screen Shot 2023-04-19 at 2 20 02 PM

We want to communicate that these images are not intended for production use. Perhaps we could change "stable" to "alpha" (I'm open to suggestions.)

This should be a trivial change so I'm giving this a size of 3.

Tags appear at https://hub.docker.com/r/gdcc/base/tags

Screen Shot 2023-04-19 at 2 22 10 PM

### PR DESCRIPTION
**Pull request by poikilotherm**
**What this PR does / why we need it**:

This pull request will enable pushing Dataverse application images from the stable master branch, nightly builds from the unstable development branch and feature builds from pull requests. It will also enable scheduled rebuilds to keep the important/supported tags up to date.

**Which issue(s) this PR closes**:

- Closes #9444
- Closes #9539 

**Special notes for your reviewer**:
This is not yet ready for review. We will need to investigate the correct workflow setups in a separate pet project.

Here's the fancy pet project. Works like a charm. https://github.com/poikilotherm/test-image-push-flows

**Suggestions on how to test this**:
None yet.

**Does this PR introduce a user interface change? If mockups are available, please link/include them here**:
No.

**Is there a release notes update needed for this change?**:
Yes. Will be included once it works.

**Additional documentation**:
None.

### CODE REVIEW

**Comment by pdurbin** [comment]
@poikilotherm @GPortas and I have been [discussing](https://dataverse.zulipchat.com/#narrow/stream/375812-containers/topic/push.20to.20registry/near/342138502) this PR and decided to give it a size of 3.

At the sprint planning kickoff just now, I advocated for a column in the global backlog for containerization (and mentioned it's a topic for the working group meeting next week). For now, I put it at the top of the community column.

**Comment by poikilotherm** [comment]
I added a link to the [pet project](https://github.com/poikilotherm/test-image-push-flows) in the description.

We can simply copy the scripts from there, rename the targes/modules and drop it in here. Probably should add some docs to our container guides how that pipeline works.

File: .github/workflows/container_base_push.yml
Comment (pdurbin) [inline .github/workflows/container_base_push.yml:84]: As discussed, can we please change this from "stable" to something else? In the issue below I'm suggesting "alpha":

- #9539

**Review by pdurbin — COMMENTED** [review]
We agreed to switch from "stable" to something else.

Also, what else is need to move this pull request from "draft" to "ready for review"?

**Comment by poikilotherm** [comment]
@pdurbin I just moved this to ready for review. Someone should take a look at the text and docs. It seems the workflow itself does what we expect it to do. We will only learn about mistakes with the Docker Hub push after merging, when a push build fails.

**Comment by poikilotherm** [comment]
@GPortas @MellyGray  as you'd be the folks mostly reusing this feature of PR-based images, please let me know if having a bot generated comment like shown above is helpful or not.

File: .github/workflows/container_app_push.yml
Comment (pdurbin) [inline .github/workflows/container_app_push.yml:11]: This should be master instead of main, right?

File: .github/workflows/container_app_push.yml
Comment (pdurbin) [inline .github/workflows/container_app_push.yml:83]: This also should be master instead of main, right?

File: .github/workflows/container_app_push.yml
Comment (pdurbin) [inline .github/workflows/container_app_push.yml:85]: This should be alpha instead of stable, right? For now, we'd like to avoid terms like stable.

File: src/main/docker/README.md
Comment (pdurbin) [inline src/main/docker/README.md:3]:
```suggestion
The "application image" offers you a deployment-ready Dataverse application running on the underlying
```

File: src/main/docker/README.md
Comment (pdurbin) [inline src/main/docker/README.md:28]:
```suggestion
You are welcome to ask questions at https://chat.dataverse.org and https://groups.google.com/g/dataverse-community
```

**Review by pdurbin — COMMENTED** [review]
Look great! I left a few comments and suggestions.

File: .github/workflows/container_app_push.yml
Comment (poikilotherm) [reply .github/workflows/container_app_push.yml]: Good catch, done.

File: .github/workflows/container_app_push.yml
Comment (poikilotherm) [reply .github/workflows/container_app_push.yml]: Good catch, done.

File: .github/workflows/container_app_push.yml
Comment (poikilotherm) [reply .github/workflows/container_app_push.yml]: Good catch, done.

File: src/main/docker/README.md
Comment (poikilotherm) [reply src/main/docker/README.md]: Good catch, done.

File: src/main/docker/README.md
Comment (poikilotherm) [reply src/main/docker/README.md]: Good catch, done.

**Comment by MellyGray** [comment]
> @GPortas @MellyGray as you'd be the folks mostly reusing this feature of PR-based images, please let me know if having [a bot generated comment like shown above](https://github.com/IQSS/dataverse/pull/9447#issuecomment-1520129701) is helpful or not.

The bot is really cool! I haven't use this PR image feature yet, but it seems really useful, thanks!

**Review by pdurbin — APPROVED** [review]
Looks good! API tests are passing. Let's give it a whirl!

**Pull request merged**
